app/models/factory.rb

```diff
@@ -0,0 +1,7 @@
+class Factory < ActiveRecord::Base
+  belongs_to :manufacturer
+
+  validates :city, uniqueness: true, presence: true
```

Comment:
don't validate uniqueness
...but if you did, it should be per-manufacturer
validates :city, uniqueness: { scope: :manufacturer_id }